homeassistant/components/ollama_conversation/config_flow.py

```diff
@@ -0,0 +1,155 @@
+"""Config flow for Ollama conversation integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import sys
+from types import MappingProxyType
+from typing import Any
+
+import httpx
+import ollama
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    TemplateSelector,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import (
+    CONF_MAX_HISTORY,
+    CONF_MODEL,
+    CONF_PROMPT,
+    DEFAULT_PROMPT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    MAX_HISTORY_NO_LIMIT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.URL)
+        ),
+        vol.Required(CONF_MODEL): cv.string,
+        vol.Required(
+            CONF_PROMPT, description={"suggested_value": DEFAULT_PROMPT}
+        ): TemplateSelector(),
```

Comment:
I don't think that we should ask these in the first form. It should just be the URL. For model, maybe has 2nd step when we could query available models ? Or maybe we just defaullt to llama2 ?

Reply:
Ollama will only report models that you've explicitly pulled down, so a fresh install will report an empty list.
For some bizarre reason, it will also not automatically pull down a model on first use! So any default here will be very wrong for some people, unfortunately.